### PR TITLE
FIX: mixin classes that have components must inherit from Device

### PR DESCRIPTION
**Pull request by tacaswell**
This is likely a side-effect of changing from using a full meta-class
to `__init_subclass__`.

### CODE REVIEW

**Comment by JunAishima** [comment]
can you remove the device.BlueskyInterface import? This is causing the test to fail.

File: ophyd/device.py [hunk 1]
```diff
@@ -1205,7 +1205,14 @@ def _instantiate_component(self, attr):
             # Initial access of signal
             cpt = self._sig_attrs[attr]
         except KeyError:
-            raise AttributeError(f'Component does not exist: {attr}') from None
+            raise RuntimeError(
+                f'Component {attr!r} exists at the Python level and '
+                'the has triggered the `_instantiate_component` '
```
Comment (danielballan) [inline ophyd/device.py:1210]: Missing a word here, can't quite tell which one.

Comment (JunAishima) [reply ophyd/device.py:1210]: there should be no first "the" in the second line?

Comment (prjemian) [reply ophyd/device.py:1210]: Not that, still wouldn't make sense.  Perhaps the missing word is _Device_? As in 

> Component {attr!r} exists at the Python level and the Device has triggered the `_instantiate_component`

Comment (prjemian) [reply ophyd/device.py:1210]: or 

> ... the caller has triggered ...

Comment (JunAishima) [reply ophyd/device.py:1210]: @tacaswell would you like to fix up the second half-sentence? thanks, will pass the review immediately once it is the way you want it (and looks like a complete sentence)

Comment (tacaswell) [reply ophyd/device.py]: Y'all knew what I meant...

File: ophyd/device.py [hunk 2]
```diff
@@ -1206,10 +1206,10 @@ def _instantiate_component(self, attr):
             cpt = self._sig_attrs[attr]
         except KeyError:
             raise RuntimeError(
-                f'Component {attr!r} exists at the Python level and '
-                'the has triggered the `_instantiate_component` '
+                f'The Component {attr!r} exists at the Python level and '
+                'has triggered the `_instantiate_component` '
                 'code path on Device, but has not been registered with '
-                'the Component management machinery.  This maybe due to '
+                'the Component management machinery in Device.  This maybe due to '
```
Comment (danielballan) [inline ophyd/device.py:1212]:
```suggestion
                'the Component management machinery in Device.  This may be due to '
```

**Pull request merged**